### PR TITLE
[mosq]:  Bump: v2.0.20 -> 2.0.20~1

### PR DESCRIPTION
**Pull request by david-cermak**
* Added missing API docs
* Replaced local port compat headers/sources and used `sock_utils` instead
* Added a CI job to perform some consistency checks:
  - all version numbers are the same
  - generated api.md is up to date

### CODE REVIEW

**Comment by david-cermak** [comment]
Another reason is that the version 2.0.20 failed to upload correctly to the registry for some reason: https://components.espressif.com/components/espressif/mosquitto/versions/2.0.20?language=en 
(the package is corrupted and cannot be downloaded)
--
The plan is to yank the damaged version and use `2.0.20~1` instead.

File: components/mosquitto/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## [2.0.20~1](https://github.com/espressif/esp-protocols/commits/mosq-v2.0.20_1)
```
Comment (kostaond) [inline components/mosquitto/CHANGELOG.md:3]: You also removed multiple files. Is it something users should be informed about?

**Review by kostaond — APPROVED** [review]
Just one minor comment, otherwise LGTM!

File: components/mosquitto/CHANGELOG.md (hunk 1 above)
Comment (david-cermak) [reply components/mosquitto/CHANGELOG.md:3]: Yes, you're right. I removed those files after the bump commit so it didn't appear in the changelog.

Updated, Thanks!

**Pull request merged**